src/api/cat.js

```diff
@@ -1,7 +1,32 @@
 'use strict'
 
-const argCommand = require('../cmd-helpers').argCommand
+const bs58 = require('bs58')
+const isIPFS = require('is-ipfs')
+const promisify = require('promisify-es6')
 
 module.exports = (send) => {
-  return argCommand(send, 'cat')
+  const cat = promisify((multihash, callback) => {
+    try {
+      multihash = cleanMultihash(multihash)
+    } catch (err) {
+      return callback(err)
+    }
+    send('cat', multihash, null, null, function (err, result) {
+      if (err) {
+        return callback(err)
+      }
+      return callback(null, result)
+    })
+  })
+  return cat
+}
+
+function cleanMultihash (multihash) {
```

Comment:
I think this is defined in `src/api/object.js` too. Could you toss it into a module in `src/`?